x/oracle/keeper/aggregator/aggregator_calculator.go

```diff
@@ -0,0 +1,168 @@
+package aggregator
+
+import (
+	"math/big"
+
+	"github.com/ExocoreNetwork/exocore/x/oracle/keeper/common"
+	"github.com/ExocoreNetwork/exocore/x/oracle/types"
+)
+
+type confirmedPrice struct {
+	sourceId  int32
+	detId     string
+	price     *big.Int
+	timestamp string
+}
+
+// internal struct
+type priceAndPower struct {
+	price *big.Int
+	power *big.Int
+}
+
+// for a specific DS round, it could have multiple values provided by different validators(should not be true if there's no malicious validator)
+type roundPrices struct { //0 means NS
+	detId     string
+	prices    []*priceAndPower
+	price     *big.Int
+	timestamp string
+	// confirmed bool
+}
+
+// udpate priceAndPower for a specific DSRoundID, if the price exists, increase its power with provided data
+// return confirmed=true, when detect power exceeds the threshold
+func (r *roundPrices) updatePriceAndPower(pw *priceAndPower, totalPower *big.Int) (updated bool, confirmed bool) {
```

Comment:
In the updatePriceAndPower method, when new prices and weights are added to roundPrices, if any exceptions are encountered (e.g., if pw.price or pw.power is nil), the code doesn't perform error handling